### PR TITLE
Document surface_collision.h

### PR DESCRIPTION
**Pull request by Multi-Volt**
Add documentation to the functions of surface_collision.h

### CODE REVIEW

File: autogen/lua_definitions/functions.lua [hunk 1]
```diff
@@ -9332,6 +9332,7 @@ end
 --- @param y number
 --- @param z number
 --- @return number
+--- Finds the height of the highest ceiling above a given position (x, y, z). If no ceiling is found, returns the default height limit of 20000.
```
Comment (PeachyPeachSM64) [inline autogen/lua_definitions/functions.lua]: maybe replace "20000" by "`gLevelValues.cellHeightLimit` (default 20000)" so the reader knows that they can control this value

File: autogen/lua_definitions/functions.lua [hunk 2]
```diff
@@ -9340,26 +9341,30 @@ end
 --- @param y number
 --- @param z number
 --- @return number
+--- Finds the height of the highest floor below a given position (x, y, z). If no floor is found, returns the default floor height of -11000.
```
Comment (PeachyPeachSM64) [inline autogen/lua_definitions/functions.lua]: same here, but with "`gLevelValues.floorLowerLimit` (default -11000)"

File: autogen/lua_definitions/functions.lua [hunk 3]
```diff
@@ -9340,26 +9341,30 @@ end
...
 function find_floor_height(x, y, z)
     -- ...
 end
 
 --- @param x number
 --- @param z number
 --- @return number
+--- Finds the height of the poison gas at a given position (x, z), if the position is within a gas region. If no gas is found, returns the default height of -11000.
```
Comment (PeachyPeachSM64) [inline autogen/lua_definitions/functions.lua]: same here, but with "`gLevelValues.floorLowerLimit` (default -11000)"

File: autogen/lua_definitions/functions.lua [hunk 4]
```diff
@@ -9340,26 +9341,30 @@ end
...
 --- @return integer
+--- Detects wall collisions at a given position and adjusts the position based on the walls found. Returns the number of wall collisions detected.
 function find_wall_collisions(colData)
     -- ...
 end
 
 --- @param x number
 --- @param z number
 --- @return number
+--- Finds the height of water at a given position (x, z), if the position is within a water region. If no water is found, returns the default height of -11000.
```
Comment (PeachyPeachSM64) [inline autogen/lua_definitions/functions.lua]: same here, but with "`gLevelValues.floorLowerLimit` (default -11000)"

File: autogen/lua_definitions/functions.lua (hunk 1 above)
Comment (Multi-Volt) [reply autogen/lua_definitions/functions.lua]: Fixed this

File: autogen/lua_definitions/functions.lua (hunk 2 above)
Comment (Multi-Volt) [reply autogen/lua_definitions/functions.lua]: Fixed this

File: autogen/lua_definitions/functions.lua (hunk 3 above)
Comment (Multi-Volt) [reply autogen/lua_definitions/functions.lua]: Fixed this

File: autogen/lua_definitions/functions.lua (hunk 4 above)
Comment (Multi-Volt) [reply autogen/lua_definitions/functions.lua]: Fixed this

**Pull request merged**